app/models/ems_refresh.rb

```diff
@@ -171,7 +171,13 @@ def self.queue_merge(targets, ems, create_task = false)
 
     # Items will be naturally serialized since there is a dedicated worker.
     MiqQueue.put_or_update(queue_options) do |msg, item|
-      targets = msg.nil? ? targets : (msg.args[0] | targets)
+      # Handle merging targets for old queue items that still have in args[0]
+      if msg && msg.args[0]
+        msg.data.concat(msg.args.shift)
+        msg.args = []
+      end
+
+      targets = msg.nil? || msg.data.nil? ? targets : msg.data.concat(targets)
```

Comment:
Same here, this can go back to `msg.nil? ? targets : ...` once we get https://github.com/ManageIQ/manageiq-schema/pull/107 in